bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java

```diff
@@ -113,9 +121,21 @@ public synchronized void start(String thingName, ShellyThingConfiguration config
             coapServer.start(config.localIp, this);
             statusClient = new CoapClient(completeUrl(config.deviceIp, COLOIT_URI_DEVSTATUS))
                     .setTimeout((long) SHELLY_API_TIMEOUT_MS).useNONs().setEndpoint(coapServer.getEndpoint());
+            @Nullable
+            Endpoint endpoint = null;
+            if (statusClient != null) {
+                endpoint = statusClient.getEndpoint();
+            }
+            if ((endpoint == null) || !endpoint.isStarted()) {
+                logger.warn("{}: Unable to initialize CoAP access (network error)", thingName);
+                throw new ShellyApiException("Network initialization failed");
+            }
             discover();
+        } catch (SocketException e) {
+            logger.warn("{}: Unable to initialize CoAP access (socket exception) - {}", thingName, e.getMessage());
+            throw new ShellyApiException("Network error", e);
         } catch (UnknownHostException e) {
-            logger.debug("{}: CoAP Exception", thingName, e);
+            logger.info("{}: CoAP Exception (Unknown Host)", thingName, e);
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug or warn. Also, it seems a bit unreasonable to log the stack trace when the network fails.

Reply:
changed